### ISSUE TITLE
Support TTY-styled chat connect; compatible with IRC Clients

### ISSUE THREAD

**Issue by ghost** [RocketChat/Rocket.Chat#1259]
Are there any plans to add support for connecting to RocketChat from IRC clients? (Like connecting to irc.slack.com through XChat)

Support for buffering unread messages and also notifications for edited/deleted messages would be a nice addition. (Slack doesn't do either of these; there is no way of knowing if messages are edited/deleted, and new messages are lost when disconnected.)

This would be a very nice feature to have, IRC clients take up very little resources.
##

**Comment by pierreozoux** [comment]
Is it a duplicate of https://github.com/RocketChat/Rocket.Chat/issues/216 ?

**Comment by edwardmue** [comment]
No.

The existing rocketchat-irc is for connecting to an existing IRC server, not acting as an IRC server for clients.

If you read what was said in https://github.com/RocketChat/Rocket.Chat/issues/216#issuecomment-143717401 and compared that to my feature request, you would already know that this isn't a duplicate.

**Comment by Sing-Li** [comment]
No.  There is no plan to do so.

Two major engineering steps required:

1.   create and implement a TTY styled protocol that covers everything we do reactively 
2.   make that protocol work for IRC clients

PRs welcomed from any dev or groups that need this badly.

**Comment by toxicgumbo** [comment]
With the answer being "no plan to do so", no developer interest expressed, and no ongoing development, I recommend this issue be closed.  If interest changes, this issue can serve as a placemarker for a new initiative or can be reopened.  As a personal note, I can see the value in using Rocket.Chat from external clients, but given its evolving codebase as an independent and "modern" environment, I can't see developers making Rocket.Chat a second-class citizen to IRC.

**Comment by purp** [comment]
Howdy! Please take this as developer interest, though not sure I've got the time or the skills required.

Separately, this isn't an issue of being a second-class citizen to IRC; this is an issue of meeting folks where they are (many of whom are in IRC for other reasons) and not requiring them to adopt a separate client when they've already got one deeply rooted in their workflow.

I suppose it's arguable that we should connect a whole IRC server to our Rocket instance a la #216, but in our specific case it's not advisable nor practical.

**Comment by LongLiveCHIEF** [comment]
I second @purp's comments. I do _everything_ in a terminal these days, and having to exit fullscreen, or switch to different workspace to chat/view messages with my team in a web/desktop client is a bit disruptive. (Programmer's, ADD, etc...).

It's not a must have, but it's definitely a "would love to have".  It would also add the advantage of being easily used from my CI/CD tools without even requiring webhooks, as well as being easily dockerized.

I've got enough interest in it, that before I found this issue, I was exploring the idea of creating a client myself.  Figured I should check to see if one was already in progress.  I don't think it's something I would want to do alone, but if several people wanted to get together to work on it, I would happily contribute to the effort.
